File: Laborator/CSharp/AgentieTurism/Tests/EmployeeRepositoryTest.cs
using System;
using System.Collections.Generic;
using System.IO;
using NUnit.Framework;
using AgentieTurism.Models;
using AgentieTurism.Repository;

namespace AgentieTurism.Tests
{
    [TestFixture]
    public class EmployeeRepositoryTest
    {
        private static IEmployeeRepository employeeRepo;
        private static Employee employee;

        [OneTimeSetUp]
        public void Setup()
        {
            //var props = new Dictionary<string, string>();
            //foreach (var line in File.ReadLines("bd.config"))
            //{
            //    var parts = line.Split('=');
            //    if (parts.Length == 2)
            //        props[parts[0].Trim()] = parts[1].Trim();
            //}

            //employeeRepo = new EmployeeDbRepository(props);
            employeeRepo = new EmployeeDbRepository();
        }

        [Test, Order(1)]
        public void TestAddEmployee()
        {
            employee = new Employee("Pop Iosua", "popiosua@email.com", "password123");
            employeeRepo.Add(employee);
            Assert.That(employee.Id, Is.Not.Null);
        }

        [Test, Order(2)]
        public void TestFindEmployeeById()
        {
            var foundEmployee = employeeRepo.FindOne(employee.Id);
            Assert.That(foundEmployee, Is.Not.Null);
            Assert.That(foundEmployee.Name, Is.EqualTo("Pop Iosua"));
        }

        [Test, Order(3)]
        public void TestUpdateEmployee()
        {
            employee.Email = "pop.iosua@email.com";
            employeeRepo.Update(employee.Id, employee);
            var updatedEmployee = employeeRepo.FindOne(employee.Id);
            Assert.That(updatedEmployee.Email, Is.EqualTo("pop.iosua@email.com"));
        }

        [Test, Order(4)]
        public void TestFindAllEmployees()
        {
            var employees = employeeRepo.FindAll();
            Assert.That(employees, Is.Not.Empty);
        }

        [Test, Order(5)]
        public void TestLogIn()
        {
            var loggedIn = employeeRepo.LogIn("pop.iosua@email.com", "password123");
            Assert.That(loggedIn, Is.Not.Null);
            Assert.That(loggedIn.Name, Is.EqualTo("Pop Iosua"));
        }

        [Test, Order(6)]
        public void TestDeleteEmployee()
        {
            employeeRepo.Delete(employee.Id);
            Assert.That(employeeRepo.FindOne(employee.Id), Is.Null);
        }
    }
}

File: Laborator/Java/AgentieTurism/src/test/java/EmployeeRepositoryTest.java
import org.junit.jupiter.api.*;
import ro.mpp2024.domain.Employee;
import ro.mpp2024.repository.EmployeeDBRepository;
import ro.mpp2024.repository.EmployeeRepository;

import java.io.FileReader;
import java.io.IOException;
import java.util.List;
import java.util.Properties;

import static org.junit.jupiter.api.Assertions.*;

@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
public class EmployeeRepositoryTest {
    private static EmployeeRepository employeeRepo;
    private static Employee employee;

    @BeforeAll
    static void setup() {
        Properties props = new Properties();
        try {
            props.load(new FileReader("bd.config"));
        } catch (IOException e) {
            fail("Cannot find bd.config: " + e);
        }
        employeeRepo = new EmployeeDBRepository(props);
    }

    @Test
    @Order(1)
    void testAddEmployee() {
        employee = new Employee("Pop Iosua", "popiosua@email.com", "password123");
        employeeRepo.add(employee);
        assertNotNull(employee.getId());
    }

    @Test
    @Order(2)
    void testFindEmployeeById() {
        Employee foundEmployee = employeeRepo.findOne(employee.getId());
        assertNotNull(foundEmployee);
        assertEquals("Pop Iosua", foundEmployee.getName());
    }

    @Test
    @Order(3)
    void testUpdateEmployee() {
        employee.setEmail("pop.iosua@email.com");
        employeeRepo.update(employee.getId(), employee);
        Employee updatedEmployee = employeeRepo.findOne(employee.getId());
        assertEquals("pop.iosua@email.com", updatedEmployee.getEmail());
    }

    @Test
    @Order(4)
    void testFindAllEmployees() {
        List<Employee> employees = (List<Employee>) employeeRepo.findAll();
        assertFalse(employees.isEmpty());
    }

    @Test
    @Order(5)
    void testLogIn() {
        Employee loggedIn = employeeRepo.logIn("pop.iosua@email.com", "password123");
        assertNotNull(loggedIn);
        assertEquals("Pop Iosua", loggedIn.getName());
    }

    @Test
    @Order(6)
    void testDeleteEmployee() {
        employeeRepo.delete(employee.getId());
        assertNull(employeeRepo.findOne(employee.getId()));
    }
}
